### DISCUSSION TITLE
Pour section's header and footer from previous section / with default one

### DISCUSSION THREAD

**Discussion by gilleschapron** [davidgohel/officer#558] [Q&A]
Hi @davidgohel and officer's user,

I have to produce a document with a template (just one page with a header - a logo - and a footer - a table with title, date, page etc.).
When I do a `body_end_block_section()`: 
- next section keep the header/footer of my template
- but the previous have not (`prop_section(header_default)` and  `prop_section(footer_default)` are NULL by default)

Is it possible to: 
- keep headers and footers through sections
- or after I read my template, to retrieve my header and footer and call them into `prop_section(xxx_default)` 

I try naively with no success:

```r
doc <- read_docx(path = "template.docx"))

test_head <- block_list(doc$headers)
test_foot <- block_list(doc$footers)

ps_portrait <- prop_section(
  page_size = page_size(orient = "portrait"),
  header_default = test_head,
  footer_default = test_foot
)

doc <- doc |>
  body_end_block_section(
    value = block_section(property = ps_portrait)
  )
```

Thanks

**Comment by FGL-Linear** [comment]
I've got the same problem. Does anyone know any solution to this?

related: [https://github.com/davidgohel/officer/issues/465#issuecomment-1399341984](url)

**Comment by gilleschapron** [comment]
Hello, 

The new parameters of `print`: `copy_header_refs` and `copy_footer_refs` are really close to solve my problem!
But: 
- template margins are not recovered : it's ok, I can easily adjust margins with `prop_section(page_margins(...))`
- sections are set to ‘differentiate odd and even pages’ whereas template configuration is not: I can't find how to disable this setting 🫤, how can I do that ?

**Comment by davidgohel** [comment]
Fixed in https://github.com/davidgohel/officer/issues/665
